include/pstl_offload/internal/usm_memory_replacement.h

```diff
@@ -0,0 +1,246 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_PSTL_OFFLOAD_INTERNAL_USM_MEMORY_REPLACEMENT_H
+#define _ONEDPL_PSTL_OFFLOAD_INTERNAL_USM_MEMORY_REPLACEMENT_H
+
+#ifndef __SYCL_PSTL_OFFLOAD__
+#error "-fsycl-pstl-offload option should be passed to the compiler to use this header"
+#endif
+
+#include <atomic>
+#include <cstdlib>
+#include <cassert>
+#include <cerrno>
+
+#include <sycl/sycl.hpp>
+
+#include <oneapi/dpl/execution>
+
+#include "usm_memory_replacement_common.h"
+
+namespace __pstl_offload {
+
+static std::atomic<sycl::device*> __active_device = nullptr;
```

Comment:
Is `__active_device` static because different offload policies are supported for different TU? Was it tested?

Reply:
Yes, it was done for building different TUs with different `-fsycl-pstl-offload` values and offload different TUs to different devices. An unfortunately we don't have a regular test for this behavior as part of this PR.
We discussed internally that we need also a test for releasing the memory in different translation unit and that we need to adapt our oneDPL testing infrastructure to build tests containing several TUs. I guess test for building different TUs with different `-fsycl-pstl-offload` can be also added as part this adaptation activity.